### PR TITLE
[Crash] Add additional raid integrity checks on Bot Spawn.

### PR DESCRIPTION
**Pull request by Aeadoin**
This PR adds an additional integrity check when we spawn Bots to ensure we don't have an invalid Raid state.

### CODE REVIEW

File: zone/raids.cpp [hunk 1]
```diff
@@ -1148,8 +1148,8 @@ void Raid::SendRaidAddAll(const char *who)
 			auto ram = (RaidAddMember_Struct*)outapp->pBuffer;
 			ram->raidGen.action = raidAdd;
 			ram->raidGen.parameter = m.group_number;
-			strcpy(ram->raidGen.leader_name, m.member_name);
-			strcpy(ram->raidGen.player_name, m.member_name);
+			strn0cpy(ram->raidGen.leader_name, m.member_name, 64);
+			strn0cpy(ram->raidGen.player_name, m.member_name, 64);
```
Comment (Kinglykrab) [inline zone/raids.cpp:1152]: Why not use `sizeof(ram->raidGen.leader_name)` here?

**Review by Kinglykrab — APPROVED** [review]

File: zone/raids.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/raids.cpp:1152]: You didn’t change this.

File: zone/raids.cpp [hunk 2]
```diff
@@ -1361,27 +1361,25 @@ void Raid::SendGroupUpdate(Client *to)
...
 		return;
 	}
 
+	int i = 0;
 	for (const auto& m : members) {
 		if (m.group_number == grp && strlen(m.member_name) > 0) {
 			if (m.is_group_leader) {
 				strn0cpy(gu->leadersname, m.member_name, 64);
 				if (strcmp(to->GetName(), m.member_name) != 0) {
 					strn0cpy(gu->membername[i], m.member_name, 64);
```
Comment (Kinglykrab) [inline zone/raids.cpp:1376]: sizeof here and above?

File: zone/raids.cpp [hunk 3]
```diff
@@ -1361,27 +1361,25 @@ void Raid::SendGroupUpdate(Client *to)
...
+					++i;
 				}
 			}
-			else {
-				if (strcmp(to->GetName(), m.member_name) != 0) {
-					strn0cpy(gu->membername[i], m.member_name, 64);
-					i++;
-				}
+			else if (strcmp(to->GetName(), m.member_name) != 0) {
+				strn0cpy(gu->membername[i], m.member_name, 64);
```
Comment (Kinglykrab) [inline zone/raids.cpp:1381]: sizeof?

File: zone/raids.cpp (hunk 1 above)
Comment (Aeadoin) [reply zone/raids.cpp:1152]: oops did the wrong line

**Pull request merged**